hw03/hw03.Rmd

```diff
@@ -0,0 +1,252 @@
+---
+title: "hw03"
+author: "Diego A. Dulanto"
+date: "2/18/2021"
+output:
+    html_document: default
+    github_document: default
+    keep_md: TRUE
+    theme: paper
+---
+
+# Homework 03
+
+```{r setup}
+library(dplyr)
+library(tidyr)
+library(tibble)
+library(naniar)
+library(scales)
+library(ggplot2)
+library(gridExtra)
+library(tidyverse)
+```
+
+You will analyze data looking at the relationship between **green reputation** and three personality traits--**compassion**, **intellectual curiosity**, and **openness to experiences**.
+The dataset includes data from **students** and **non-students**.
+
+```{r}
+dictionary <- readr::read_csv("green_dictionary.csv")
+green_data <- readr::read_csv("green_data.csv")
+View(dictionary)
+View(green_data)
+```
+
+
+For your assignment, do the following.
+
+  1. Inspect the item responses (e.g., with graphs or by summarizing distinct values). 
+     Is anything unusual?
+```{r}
+green_data %>% 
+  summarise(
+    across(
+      c(green1:student),
+      list(Max = ~ max(.x, na.rm = TRUE),
+           Min = ~ min(.x, na.rm = TRUE))))
+
+filter(green_data, id == 5549)
+
+dictionary %>% 
+  summarize(dictionary)
+
+green_data %>% 
+  ggplot() +
+  geom_jitter(aes(x = "id", y = "green"))
+
+#There are -99s in the data that needs to be recoded
+
+green_recode <-
+  green_data %>%
+  replace_with_na_all(~.x == -99) %>%
+  mutate(
+    student = recode(student, "1" = "nonstudent", "2" = "student"))
+
+```     
+
+  2. Compute total scores for the four scales. 
+     Recode variables as needed.
+```{r}
+reversed <- dictionary %>%
+  filter(
+    Keying == -1 | Keying == -2) %>%
+  pull(Item)
+
+reversed
+
+green_reversed <-green_recode %>%
+  mutate(
+    across(
+      all_of(reversed),
+           ~ recode(.x, 
+                    "5" = 1, 
+                    "4" = 2, 
+                    "3" = 3, 
+                    "2" = 4, 
+                    "1" = 5)))
```

Comment:
Very neat code. You could also use ```~6 - .x```. Also, you may want to change the names of your recoded items